python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -837,17 +837,23 @@ def default_job_io_manager(init_context: "InitResourceContext"):
     # support overriding the default io manager via environment variables
     module_name = os.getenv("DAGSTER_DEFAULT_IO_MANAGER_MODULE")
     attribute_name = os.getenv("DAGSTER_DEFAULT_IO_MANAGER_ATTRIBUTE")
+    silence_failures = os.getenv("DAGSTER_DEFAULT_IO_MANAGER_SILENCE_FAILURES")
+
     if module_name and attribute_name:
         from dagster._core.execution.build_resources import build_resources
 
-        module = importlib.import_module(module_name)
-        attr = getattr(module, attribute_name)
-        check.invariant(
-            isinstance(attr, IOManagerDefinition),
-            "DAGSTER_DEFAULT_IO_MANAGER_MODULE and DAGSTER_DEFAULT_IO_MANAGER_ATTRIBUTE must specify an IOManagerDefinition",
-        )
-        with build_resources({"io_manager": attr}) as resources:
-            return resources.io_manager
+        try:
+            module = importlib.import_module(module_name)
+            attr = getattr(module, attribute_name)
+            check.invariant(
+                isinstance(attr, IOManagerDefinition),
+                "DAGSTER_DEFAULT_IO_MANAGER_MODULE and DAGSTER_DEFAULT_IO_MANAGER_ATTRIBUTE must specify an IOManagerDefinition",
+            )
+            with build_resources({"io_manager": attr}) as resources:
+                return resources.io_manager
+        except (ModuleNotFoundError, AttributeError):
+            if not silence_failures:
+                raise
 
```

Comment:
is there a way to log a warning here

Reply:
outside of using warnings.warn there's no context object to route with.